### PR TITLE
[v2] Add range and Go symbol deletion tools

### PR DESCRIPTION
**Pull request by SivanCola**
## Summary
Add two writer tools for deletion workflows: `delete_range` for exact text-anchor range deletion and `delete_symbol` for Go AST-based symbol deletion.

## Root Cause
Large deletions and symbol removals are awkward with small text replacement tools and can be error-prone when done manually.

## Technical Approach
Implement a previewable `delete_range` tool with unique start/end anchors and a previewable `delete_symbol` tool that parses Go files and removes selected declarations.

## Focused Optimization Points
- Integrates both tools with writer confinement and workspace-scoped tool construction.
- Provides unified diff output for review.
- Adds tests for duplicates, missing anchors, CRLF handling, methods, types, and preview behavior.

## Verification
Not run during this publishing pass. Draft note: this needs careful review of AST edge cases such as grouped declarations, comments, formatting, and multi-symbol specs.

### CODE REVIEW

File: internal/tool/builtin/delete_range.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		keep = append(keep, lines[:startLine]...)
+		keep = append(keep, lines[endLine+1:]...)
+	} else {
+		keep = append(keep, lines[:startLine+1]...)
+		keep = append(keep, lines[endLine:]...)
+	}
+	newContent := strings.Join(keep, lineSep)
+	// Preserve trailing newline if original had one.
+	if strings.HasSuffix(original, lineSep) && !strings.HasSuffix(newContent, lineSep) {
+		newContent += lineSep
```
[the remark this answers is not in the document]
Comment (SivanCola) [reply internal/tool/builtin/delete_range.go]: Fixed in d3571537.\n\nWhat changed:\n- delete_range now produces an empty file when the selected range removes all content, including newline-terminated single-line files.\n\nRoot cause:\n- The newline preservation path re-added a trailing newline even after the whole file was deleted.\n\nVerification:\n- `gofmt -l . | grep -v '^desktop/'` passed with no output.\n- `go test ./internal/tool/builtin` passed.

File: internal/tool/builtin/delete_symbol.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+					kind := "var"
+					if d.Tok == token.CONST {
+						kind = "const"
+					}
+					for _, ident := range s.Names {
+						matches = append(matches, symbolMatch{
+							name:  ident.Name,
+							kind:  kind,
+							start: ident.Pos(),
+							end:   ident.End(),
```
[the remark this answers is not in the document]
Comment (SivanCola) [reply internal/tool/builtin/delete_symbol.go]: Fixed in d3571537.\n\nWhat changed:\n- delete_symbol now rejects ambiguous multi-name const/var specs instead of deleting the entire shared declaration for one requested symbol.\n\nRoot cause:\n- Multiple names from one GenDecl were registered independently, but deletion operated on the full source line/range.\n\nVerification:\n- `gofmt -l . | grep -v '^desktop/'` passed with no output.\n- `go test ./internal/tool/builtin` passed.
